realm/realm-library/src/main/java/io/realm/RealmFieldType.java

```diff
@@ -37,7 +37,8 @@
     BINARY(4),
     UNSUPPORTED_TABLE(5),
     UNSUPPORTED_MIXED(6),
-    DATE(7),
+    UNSUPPORTED_DATE(7),
```

Comment:
I would probably just remove this together with other unsupported types. No need to expose them in the public API I think?